tests/ui/escape_analysis.stderr

```diff
@@ -12,11 +12,5 @@ error: local variable doesn't need to be boxed here
 LL | pub fn new(_needs_name: Box<PeekableSeekable<&()>>) -> () {}
    |            ^^^^^^^^^^^
 
-error: local variable doesn't need to be boxed here
```

Comment:
This is the only remaining FN introduced in this PR now. So ping @rust-lang/clippy for review.

Reply:
file an issue for it and refer to it in the test file and we can land tis